ipq40xx: add support for Telco X1 Pro

nickberry17 opened the pull request:
Telco X1 Pro is a Cat12 LTE-A Pro modem router.
Vendor firmware is based on a recent version of OpenWrt.
Flashing is possible via CLI using `sysupgrade -F -n`
The serial headers allow bootloader and console access
Serial setting: `115200  8N1`

Brief Specifications:
IPQ4019 SoC
32MB flash
512MB RAM
4x gigabit LAN
1x gigabit WAN
Dual-band Wave-2 wifi
2x SMA LTE antenna connectors
2x RP-SMA wifi antennas
1x USB 2.0 port
1x Reset button
Serial headers installed
1x Nano SIM tray
1x Quectel EM-12G LTE-A Pro modem
1x M.2 slot attached to USB 3.0
1x internal micro SD card slot

Signed-off-by: Nicholas Smith <nicholas@nbembedded.com>

mrkiko commented:
Hello!!

thanks for your work!

Is there any recovery method available for this device ? If so, could you add it to the commit message?

Thanks! :)

hauke commented on target/linux/ipq40xx/patches-5.10/901-arm-boot-add-dts-files.patch:
```diff
@@ -65,6 +65,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 +	qcom-ipq4019-rtl30vw.dtb \
 +	qcom-ipq4019-srr60.dtb \
 +	qcom-ipq4019-srs60.dtb \
++   qcom-ipq4019-x1pro.dtb \
```
Please fix the indention here

hauke commented on target/linux/ipq40xx/patches-5.4/901-arm-boot-add-dts-files.patch:
```diff
@@ -63,6 +63,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 +	qcom-ipq4019-rtl30vw.dtb \
 +	qcom-ipq4019-srr60.dtb \
 +	qcom-ipq4019-srs60.dtb \
++   qcom-ipq4019-x1pro.dtb \
```
Please fix the indention here

hauke commented on target/linux/ipq40xx/image/generic.mk:
```diff
@@ -873,6 +873,18 @@ define Device/qxwlan_e2600ac-c2
+	KERNEL_SIZE := 4096k
+	IMAGE_SIZE := 31232k
+	IMAGE/sysupgrade.bin := append-kernel | append-rootfs | pad-rootfs | append-metadata
+	DEFAULT := n
```
Why is this deactivated by default?

nickberry17 replied:
Hmm, actually, I don't understand this `DEFAULT`; could you tell me what that is for please?

hauke replied:
The `DEFAULT := n` will deactivate this board by default in the build. We use it for example when the image does not fit onto the flash in the default configuration to not build the images with build bot.

hauke commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-x1pro.dts:
```diff
@@ -0,0 +1,92 @@
+
+	mtd-mac-address = <&art 0x5006>;
+	mtd-mac-address-increment = <3>;
+};
```
Please add a new line here

nickberry17 replied:
done

hauke commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-x1pro.dtsi:
```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
```
Why do you have this additional dtsi file? Are there multiple variants of this device you plan to ad later?

nickberry17 replied:
> Are there multiple variants of this device you plan to ad later?

Yes, a 5G version

chunkeey commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-x1pro.dtsi:
```diff
@@ -0,0 +1,224 @@
+};
+
+&wifi0 {
+	status = "okay";
```
Are you planning on adding WIFI or are these placeholders?

(If so, then it looks like some stuff is missing. Might have been forgotten files maybe?)

nickberry17 replied:
I have linked these up to the calibration data

Commit:
ipq40xx: add support for Telco X1 Pro

Telco X1 Pro is a Cat12 LTE-A Pro modem router.
Vendor firmware is based on a recent version of OpenWrt.
Flashing is possible via CLI using sysupgrade -F -n
The serial headers allow bootloader and console access
Serial setting: 115200  8N1

Brief Specifications:
IPQ4019 SoC
32MB flash
512MB RAM
4x gigabit LAN
1x gigabit WAN
Dual-band Wave-2 wifi
2x SMA LTE antenna connectors
2x RP-SMA wifi antennas
1x USB 2.0 port
1x Reset button
Serial headers installed
1x Nano SIM tray
1x Quectel EM-12G LTE-A Pro modem
1x M.2 slot attached to USB 3.0
1x internal micro SD card slot

Signed-off-by: Nicholas Smith <nicholas@nbembedded.com>

nickberry17 commented:
Rebased and resolved  merge conflicts

hauke commented:
Thank you for your patch, I applied it to master in 7ea2f3d6e22b506d137cf255ae73ae498c176d1c.
